src/test/article.spec.js

```diff
@@ -304,3 +309,356 @@ describe('GET api/v1/articles', () => {
+  });
+
+  it('should successfully create an article with valid user input', async () => {
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+
+    const response = await server()
+      .post(`${ARTICLES_API}`)
+      .set('x-access-token', token)
+      .send(articleSample);
+    expect(response.status).to.equal(201);
+    expect(response.body.data.authorId).to.equal(newUser.id);
+    expect(response.body.data.title).to.equal(articleSample.title);
+    expect(response.body.data.body).to.equal(articleSample.body);
+    expect(response.body.data.image).to.equal(articleSample.image);
+    expect(response.body.data).to.have.property('slug');
+    expect(response.body.data.slug).to.be.a('string');
+    expect(response.body.data.status).to.equal('active');
+    expect(response.body.data.publishedDate).to.equal(null);
+  });
+
+  it('should test that two articles with same title are not created with the same slug', async () => {
+    const article1 = await BaseRepository.create(db.Article, articleSample);
+    const article2 = await BaseRepository.create(db.Article, articleSample2);
+
+    expect(article1.slug).to.not.equal(article2.slug);
+  });
+
+  it('should throw a 400 status code when creating an article with no title', async () => {
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+
+    const response = await server()
+      .post(`${ARTICLES_API}`)
+      .set('x-access-token', token)
+      .send(articleWithNoTitle);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal(
+      'Please provide a title for your article with minimum of 3 characters'
+    );
+  });
+
+  it('should return error if database error occurs', async () => {
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+    const findAllStub = await sinon.stub(BaseRepository, 'create');
+    findAllStub.rejects(
+      new Error(
+        'Your request cannot be processed right now, Please try again later'
+      )
+    );
+    const res = await chai
+      .request(app)
+      .post(`${ARTICLES_API}`)
+      .set('x-access-token', token)
+      .send(articleSample);
+    expect(res.status).to.equal(500);
+    expect(res.body.message).to.equal(
+      'Your request cannot be processed right now, Please try again later'
+    );
+    findAllStub.restore();
+  });
+});
+
+describe('GET /api/v1/articles/:articleId', () => {
+  beforeEach(async () => {
+    await db.Bookmark.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+    await db.Article.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should successfully get a specific article', async () => {
+    const validArticle = await BaseRepository.create(db.Article, articleSample);
+    const article = await BaseRepository.findOne(db.Article, validArticle.id);
```

Comment:
Is there a reason why you are finding one here?

Reply:
Thank you for this feedback.

Yes, this is a repetition to test and/or mimic the expected behaviour of the application.
Here, the `validArticle` is used for the route request, while the `article` is used for the assertions to test the logic implemented thereafter.